### ISSUE TITLE
[a11y] Suggestion details are not available to screen readers

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#12905]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: Monaco 0.34.1
- OS Version: Mac OS 12.6.1, Windows 10 20H2 Build 19042.2251

Steps to Reproduce:

1.  Enable screen reader (JAWS 2020, Voiceover)
2. Open Monaco https://microsoft.github.io/monaco-editor/index.html 
3. On a new line, type a code snippet to autocomplete, ie "window.a". The autosuggest overlay will appear.
    * screen reader reads name and type declaration of first selected item.
4. Toggle auto suggest details (click right chevron on highlighted item, or press ctrl+i / cmd+i)
    * Screen reader will not read information in the suggestion overlay
    * User is not notified that there is additional content on the screen
    * User is not provided information about viewing additional content
    * User cannot use arrow keys / change screen reader navigation modes to navigate to the content as these are handled by the editor and close the overlay

Expected Behavior

1. When the suggestion overlay appears, it is indicated to the user that there is a subtree item which they are able to view
2. When using a keyboard modality to navigate the suggestions list, a command is made available to users to interact with the details item
3. Screen reader users are able to perceive documentation for suggestions which is provided to sighted users
4. A proper tree-treeitem relationship is used to define the association between the autosuggest and details overlays. See: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/

In its current state, this has the fundamental issue in that it provides information to sighted users that it does not make available to screen reader users. More specifically, [WCAG 2.1 1.3.1 Level A](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) states:

> The intent of this Success Criterion is to ensure that information and relationships that are implied by visual or auditory formatting are preserved when the presentation format changes. For example, the presentation format changes when the content is read by a screen reader or when a user style sheet is substituted for the style sheet provided by the author.

The structure of the information provided, according to this criteria, is also relevant and should be conveyed to users.
